### PR TITLE
Add contributing documentation

### PR DESCRIPTION
**Pull request by ryanjdew**

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+Be sure to run the tests before submitting your pull request. PRs with failing
+tests won't be accepted.
+
+First verify that you have valid server and admin auth info in
+`username` and `password` fields in
+`nifi-nar-bundles/nifi-marklogic-bundle/nifi-marklogic-processors/src/test/java/org/apache/nifi/marklogic/processor/AbstractMarkLogicIT.java` and `nifi-nar-bundles/nifi-marklogic-bundle/nifi-marklogic-services/src/test/java/org/apache/nifi/marklogic/controller/DefaultMarkLogicDatabaseClientServiceIT.java`
+
+If compatible NiFi SNAPSHOT hasn't been installed in your local Maven yet (Grab a cup of coffee while it runs.):
+
+    $ mvn -T 2 clean install -Pcontrib-check,include-grpc,include-atlas,include-hive3 -Ddir-only | grep -v -F -f .travis-output-filters
```
Comment (dmcassel) [inline .github/CONTRIBUTING.md]: if you're running on Windows, I don't think this will work unless you add the skip-tests option

File: README.md [hunk 2]
```diff
@@ -12,15 +12,12 @@
   See the License for the specific language governing permissions and
   limitations under the License.
 -->
-[<img src="https://nifi.apache.org/assets/images/apache-nifi-logo.svg" width="300" height="126" alt="Apache NiFi"/>][nifi]
+|Apache NiFi                |  MarkLogic                |
+|:-------------------------:|:-------------------------:|
+|[<img src="https://nifi.apache.org/assets/images/apache-nifi-logo.svg" width="300" height="126" alt="Apache NiFi"/>][nifi]      |  [<img src="https://cdn1.marklogic.com/wp-content/uploads/2017/12/marklogic-logo-footer.svg" width="300" height="126" alt="MarkLogic"/>][marklogic]|
```
Comment (dmcassel) [inline README.md]: I had no idea we had a CDN.

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (ryanjdew) [reply .github/CONTRIBUTING.md]:
```suggestion
    $ mvn -T 2 clean install -Ddir-only -DskipTests | grep -v -F -f .travis-output-filters
```

File: README.md (hunk 2 above)
Comment (ryanjdew) [reply README.md]: I grabbed the link from the footer of marklogic.com.

**Pull request merged**